From aris@ruivo.org  Mon Nov 26 21:57:27 2007
From: Aristeu Rozanski <aris@ruivo.org>
Date: Mon, 12 Nov 2007 15:14:49 -0500
Subject: USB: usb_serial_console: fix command line parsing
To: linux-serial@vger.kernel.org
Cc: Greg KH <greg@kroah.com>
Message-ID: <20071112201449.GH407@cathedrallabs.org>
Content-Disposition: inline


Currently usb serial console support ignores the device and always use
ttyUSB0.

Signed-off-by: Aristeu Rozanski <arozansk@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/serial/console.c |    9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

--- a/drivers/usb/serial/console.c
+++ b/drivers/usb/serial/console.c
@@ -133,11 +133,14 @@ static int usb_console_setup(struct cons
 	}
 	co->cflag = cflag;
 
-	/* grab the first serial port that happens to be connected */
-	serial = usb_serial_get_by_index(0);
+	/*
+	 * no need to check the index here: if the index is wrong, console
+	 * code won't call us
+	 */
+	serial = usb_serial_get_by_index(co->index);
 	if (serial == NULL) {
 		/* no device is connected yet, sorry :( */
-		err ("No USB device connected to ttyUSB0");
+		err ("No USB device connected to ttyUSB%i", co->index);
 		return -ENODEV;
 	}
 
